src/plugins/ui_actions/public/cell_actions/components/inline_actions.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import type { Action } from '../../actions';
+import { ActionItem } from './cell_action_item';
+import { usePartitionActions } from '../hooks/actions';
+import { ExtraActionsPopOver } from './extra_actions_popover';
+import { ExtraActionsButton } from './extra_actions_button';
+import { CellActionExecutionContext } from './cell_actions';
+
+interface InlineActionsProps {
+  getActions: () => Promise<Action[]>;
+  actionContext: CellActionExecutionContext;
+  showTooltip: boolean;
+  showMoreActionsFrom: number;
+}
+
+export const InlineActions: React.FC<InlineActionsProps> = ({
+  getActions,
+  actionContext,
+  showTooltip,
+  showMoreActionsFrom,
+}) => {
+  const { extraActions, visibleActions } = usePartitionActions(getActions, showMoreActionsFrom);
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const togglePopOver = useCallback(() => setIsPopoverOpen((isOpen) => !isOpen), []);
+  const closePopOver = useCallback(() => setIsPopoverOpen(false), []);
+  const button = useMemo(
+    () => <ExtraActionsButton onClick={togglePopOver} showTooltip={showTooltip} />,
+    [togglePopOver, showTooltip]
+  );
+
+  return (
+    <span data-test-subj="inlineActions">
+      {visibleActions.map((action, index) => (
+        <ActionItem
+          key={`action-item-${index}`}
+          action={action}
+          actionContext={actionContext}
+          showTooltip={showTooltip}
+        />
+      ))}
+      {extraActions.length > 0 ? (
+        <ExtraActionsPopOver
+          actions={extraActions}
+          actionContext={actionContext}
+          button={button}
+          closePopOver={closePopOver}
+          isOpen={isPopoverOpen}
+        />
+      ) : null}
+    </span>
+  );
```

Comment:
wondering if these should be in `useMemo`:
```suggestion
  const inlineActions = useMemo(
    () =>
      visibleActions.map((action, index) => (
        <ActionItem
          key={`action-item-${index}`}
          action={action}
          actionContext={actionContext}
          showTooltip={showTooltip}
        />
      )),
    [actionContext, showTooltip, visibleActions]
  );

  const extras = useMemo(
    () =>
      extraActions.length > 0 ? (
        <ExtraActionsPopOver
          actions={extraActions}
          actionContext={actionContext}
          button={button}
          closePopOver={closePopOver}
          isOpen={isPopoverOpen}
        />
      ) : null,
    [actionContext, button, closePopOver, extraActions, isPopoverOpen]
  );

  return (
    <span data-test-subj="inlineActions">
      {inlineActions}
      {extras}
    </span>
  );
```